FeedAPIChallenge/Feed API/FeedItemsMapper.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  FeedItemsMapper.swift
+//  FeedAPIChallenge
+//
+//  Created by Mher Movsisyan on 08.08.21.
+//  Copyright © 2021 Essential Developer Ltd. All rights reserved.
+//
+
+import Foundation
+
+struct FeedItemsMapper {
+	struct Root: Decodable {
+		private let items: [Item]
+
+		var feedItems: [FeedImage] {
+			items.compactMap { item in
+				guard let uuid = UUID(uuidString: item.imageId),
+							let url = URL(string: item.imageURL) else { return nil }
+
+				return FeedImage(id: uuid, description: item.imageDesc, location: item.imageLoc, url: url)
+			}
```

Comment:
Following from my previous comments, by taking advantage of the Type System there will be no need to cast the values to `UUID` and `URL`: JSONDecoder will take care of that.

Also, by returning `nil` this portion of code is simply ignoring invalid items. But in the case where one item cannot be mapped the whole result should be discarded and `RemoteFeedLoader` should complete with `invalidData`.

Since we don't need to deal with `nil` values by using the correct types, there's no need for the `guard` statement nor for the `compactMap`, so this portion can be simpler:
```suggestion
			items.map { item in
				return FeedImage(id: item.image_id, description: item.image_desc, location: item.image_loc, url: item.image_url)
			}
```